File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int colonIndex = logLine.IndexOf(":");
        string modifiedMessage = logLine.Remove(0, ++colonIndex).Trim();
    
        return modifiedMessage;
    }

    public static string LogLevel(string logLine)
    {
        int levelStartIndex = logLine.IndexOf("[");
        int levelEndIndex = logLine.IndexOf("]");

        string modifiedMessage = logLine.Substring(++levelStartIndex, --levelEndIndex).ToLower();

        return modifiedMessage;
    }

    public static string Reformat(string logLine)
    {
        string modifiedMessage = Message(logLine) + $" ({LogLevel(logLine)})";
            
        return modifiedMessage;
    }
}

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        return logLine.substring(logLine.indexOf(":")+1).strip();
    }

    public static String logLevel(String logLine) {
        return logLine.substring(logLine.indexOf("[")+1,  logLine.lastIndexOf("]:")).strip().toLowerCase();
    }

    public static String reformat(String logLine) {
        return String.format("%s (%s)", message(logLine), logLevel(logLine));
    }
}
